### ISSUE TITLE
Include Sunshine

### ISSUE THREAD

**Issue by dannymate** [ublue-os/main#311]
### Describe the package

I'd like to request the package `[Sunshine](https://github.com/LizardByte/Sunshine/tree/master)` because it allows streaming to any device that has access to a Moonlight client with support for AMD, Intel, and Nvidia GPUs for hardware encoding.

It is the best way I've found to stream to any device I want. The stream quality is good and devices with Moonlight are plentiful (i.e. [Nintendo Switch Moonlight Homebrew](https://github.com/XITRIX/Moonlight-Switch)).

One issue I found and I'm not sure it affects Sunshine so much as it's a Moonlight client issue but it's worth mentioning. When the client device performs Wi-Fi scanning it causes Moonlight to hitch on Linux/GNOME. The only way I found to fix that was to set a BSSID on the client to prevent network scans. I'm unsure if this would affect the server-side in the same way.

It is available as a Flatpak however there is some setup that needs to be performed so it'd be ideal if it was baked in. Though as it is available in other packages it's just a nice to have.

### Information on the package

Installed Packages
Name         : sunshine
Version      : 0.20.0
Release      : 1
Architecture : x86_64
Size         : 48 M
Source       : sunshine-0.20.0-1.src.rpm
Repository   : @System
From repo    : @commandline
Summary      : Sunshine is a self-hosted game stream host for Moonlight.
URL          : https://app.lizardbyte.dev
License      : unknown
Description  : DESCRIPTION
             : ===========
             : 
             : This is an installer created using CPack (https://cmake.org). No
             : additional installation instructions provided.

### Image

Bazzite

**Comment by bsherman** [comment]
Sunshine is available as a Flatpak. https://flathub.org/apps/dev.lizardbyte.app.Sunshine

Have you given that a look?

In the main and nvidia repos, we try to keep the changes from upstream Silverblue/Kinoite fairly minimal, usually only adding the stuff that's harder to add, for example the nvidia kernel drivers.

**Comment by bsherman** [comment]
> When the client device performs Wi-Fi scanning it causes Moonlight to hitch on Linux/GNOME. The only way I found to fix that was to set a BSSID on the client to prevent network scans. I'm unsure if this would affect the server-side in the same way.

Interesting aside: macOS also suffers from a wifi-scanning hitch issue.

**Comment by dannymate** [comment]
> As you mention, Sunshine is available as a Flatpak. [flathub.org/apps/dev.lizardbyte.app.Sunshine](https://flathub.org/apps/dev.lizardbyte.app.Sunshine)
> 
> I'm not sure what setup could be pre-included that would be universal for users of all GPUs.
> 
> In the main and nvidia repos, we try to keep the changes from upstream Silverblue/Kinoite fairly minimal, usually only adding the stuff that's harder to add, for example the nvidia kernel drivers.

If you look at the flatpak description there's a bit of post setup people have to do. It's not much but it's not nothing.

I'm not going to push for it but it's just something I felt fit the project and would make life a tad bit easier for users.

**Comment by akdev1l** [comment]
It seems like it is adding some udev rules - we could do that. 

The way they do it in the description there isn’t very good though(they’re chown a device node to belong to $USER) so we would need to figure out how to do it properly

**Comment by bsherman** [comment]
> It seems like it is adding some udev rules - we could do that.

Agreed, adding udev rules is definitely in our scope. I haven't had a chance to look at the Sunshine flatpak that closely, but would like a take a look at that sometime. It IS something I'd like to use too.
